src/tracks/playabletrack/wavetrack/ui/WaveChannelView.cpp

```diff
@@ -1683,35 +1664,34 @@ ClipParameters::ClipParameters
 
 wxRect ClipParameters::GetClipRect(const WaveClip& clip, const ZoomInfo& zoomInfo, const wxRect& viewRect, bool* outShowSamples)
 {
-    auto srs = 1. / static_cast<double>(clip.GetRate());
-    double averagePixelsPerSample{};
-    bool showIndividualSamples{};
-    auto clipEndingAdjustemt 
-       = CalculateAdjustmentForZoomLevel(viewRect, zoomInfo, clip.GetRate(), averagePixelsPerSample, showIndividualSamples);
-    if (outShowSamples != nullptr)
-       *outShowSamples = showIndividualSamples;
-    constexpr auto edgeLeft = static_cast<ZoomInfo::int64>(std::numeric_limits<int>::min());
-    constexpr auto edgeRight = static_cast<ZoomInfo::int64>(std::numeric_limits<int>::max());
-    auto left = std::clamp(
-       zoomInfo.TimeToPosition(
-          clip.GetPlayStartTime(), viewRect.x, true
-       ), edgeLeft, edgeRight
-    );
-    auto right = std::clamp(
-       zoomInfo.TimeToPosition(
-          clip.GetPlayEndTime() - .99 * srs + clipEndingAdjustemt, viewRect.x, true
-       ), edgeLeft, edgeRight
-    );
-    if (right >= left)
-    {
-        //after clamping we can expect that left and right 
-        //are small enough to be put into int
-        return wxRect(
-           static_cast<int>(left), 
-           viewRect.y, 
-           std::max(1, static_cast<int>(right - left)), 
-           viewRect.height
-        );
+   const auto pixelsPerSecond = GetPixelsPerSecond(viewRect, zoomInfo);
+   const auto showIndividualSamples = ShowIndividualSamples(
+      viewRect, zoomInfo, clip.GetRate(), clip.GetStretchRatio(),
+      pixelsPerSecond);
+   const auto clipEndingAdjustemt =
+      CalculateAdjustmentForZoomLevel(pixelsPerSecond, showIndividualSamples);
+   if (outShowSamples != nullptr)
+      *outShowSamples = showIndividualSamples;
+   constexpr auto edgeLeft =
+      static_cast<ZoomInfo::int64>(std::numeric_limits<int>::min());
```

Comment:
Remember `min` is not `lowest` when the type is floating! -- but that's not the case here